dotcom-rendering/playwright/tests/parallel-5/liveblog.interactivity.e2e.spec.ts

```diff
@@ -0,0 +1,306 @@
+		);
+
+		await expectToNotExist(page, `[data-testid="toast"]`);
+
+		await page.evaluate(() => {
+			window.mockLiveUpdate({
+				numNewBlocks: 1,
+				html: '<p>New block 1</p>',
+				mostRecentBlockId: 'abc',
+			});
+		});
+
+		await expectToExist(page, `[data-testid="toast"]`);
+
+		const updateLocator = page
+			.getByText('1 new update')
+			.locator('visible=true');
+		await expect(updateLocator).toBeVisible();
+
+		await updateLocator.click();
+
+		expect(
+			await page.evaluate(() => window.document.location.hash),
+		).toEqual('#maincontent');
+
+		expect(
+			await page.evaluate(() => window.document.location.pathname),
+		).toEqual(
+			'/australia-news/live/2022/feb/22/australia-news-live-updates-scott-morrison-nsw-trains-coronavirus-covid-omicron-weather',
+		);
+
+		expect(
+			await page.evaluate(() => window.document.location.search),
+		).toEqual('');
+	});
+
+	test('should initially hide new blocks, only revealing them when the top of blog is in view', async ({
+		page,
+	}) => {
+		await loadPage(page, `/Article/${blogUrl}?live=true`);
+		await waitForIsland(page, 'Liveness', { waitFor: 'attached' });
+
+		await page.evaluate(() =>
+			window.scrollTo(0, document.documentElement.scrollHeight),
+		);
+
+		await page.evaluate((tweet) => {
+			window.mockLiveUpdate({
+				numNewBlocks: 1,
+				html: tweet,
+				mostRecentBlockId: 'abc',
+			});
+		}, tweetBlock);
+
+		await waitForIsland(page, 'TweetBlockComponent', {
+			waitFor: 'attached',
+		});
+
+		// have to use an attribute selector here as a numeric id is not a valid selector
+		const tweetContainerSelector =
+			'[id="46d194c9-ea50-4cd5-af8b-a51e8b15c65e"]';
+
+		await expectToNotBeVisible(page, tweetContainerSelector);
+
+		await page
+			.locator('div[data-gu-name="media"]')
+			.scrollIntoViewIfNeeded();
+
+		await expectToExist(page, tweetContainerSelector);
+	});
+
+	test('should render live score updates to a cricketblog', async ({
+		page,
+	}) => {
+		const update1ResponsePromise = stubResponse(
+			page,
+			/england-cricket-team\.json/,
+			{
+				status: 200,
+				json: {
+					scorecardUrl: '/test',
+					match: match1,
+				},
+			},
+		);
+
+		await loadPage(
+			page,
+			`/Article/https://theguardian.com/sport/live/2022/mar/27/west-indies-v-england-third-test-day-four-live?live`,
+		);
+		await waitForIsland(page, 'Liveness', { waitFor: 'attached' });
+
+		await update1ResponsePromise;
+
+		await expect(page.getByText('297 & 28 - 0 (4.5 overs)')).toBeVisible();
+
+		await expect(
+			page.getByText('204 & 120 all out (64.2 overs)'),
+		).toBeVisible();
+
+		await stubResponse(page, /england-cricket-team\.json/, {
+			status: 200,
+			json: {
+				scorecardUrl: '/test',
+				match: match2,
+			},
+		});
+
+		// The cricket data is updated every 14 seconds, so we need a longer timeout here
+		await expect(page.getByText('297 & 104 - 3 (25 overs)')).toBeVisible({
+			timeout: 30000,
+		});
+	});
```

Comment:
Is using a timeout the best way to get the result, here? Could we not simply jump a timer by 14 seconds into the future, which would trigger the update from SWR?

Reply:
Playwright does not have a method of changing the clock time. I'd prefer not to monkey patch timing functions as I think that would be flakey and error prone.

I did consider whether to keep this test because of this effective 14s wait. We've brought down the test times from ~5m to ~1m so I think an additional 14s is not unreasonable.

If this test becomes an outlier and flakey we can always remove it.